diff --git a/app/views.py b/app/views.py
index a172b60..c776133 100644
--- a/app/views.py
+++ b/app/views.py
@@ -3,6 +3,7 @@ from flask import Flask, render_template, g, request, flash, redirect, url_for
 from .forms import LoginForm, SearchForm, UserRateSubmissionsForm, UserAddStoreForm
 import sqlite3
 import time
+from flask_login import current_user
 
 
 def connect_db():
@@ -42,7 +43,7 @@ def store(store_id):
     cursor.execute('select * from stores where storeUUID=' + '"' + store_id + '"')
     row_store_info = list(cursor.fetchone())
     print(row_store_info)
-    store_uuid, latitude, longitude, display_name, strikes = row_store_info
+    store_uuid, latitude, longitude, display_name, strikes, username, password = row_store_info
 
     # Get user's rating (format: username, rating, store_uuid, timestamp)
     cursor.execute('select username, rating, timestamp from safetyRatings where storeUUID=' + store_id)
@@ -72,21 +73,27 @@ def store(store_id):
     # Convert all sql objects inside the list to list format
     store_submissions = [list(s) for s in store_submissions]
 
-    # instantiate UserRateSubmissionsForm object
-    form = UserRateSubmissionsForm(request.form)
-
-    if form.validate_on_submit():
-        flash('Update received, thank you!')
-        time.sleep(3)
-        return redirect('/store/' + store_id)
-        
-
+    if current_user.is_authenticated:
+        if current_user.get_id() == username:
+            form = StoreRateSubmissionsForm(request.form)
+        else:
+            # instantiate UserRateSubmissionsForm object
+            form = UserRateSubmissionsForm(request.form)
+        if form.validate_on_submit():
+            flash('Update received, thank you!')
+            time.sleep(2)
+            return redirect('/store/' + store_id)
 
+        return render_template("store.html", store_uuid=store_uuid, latitude=latitude, longitude=longitude,
+            display_name=display_name, strikes=strikes, safety_ratings=safety_ratings,
+            user_submissions=user_submissions, store_submissions=store_submissions, form=form)
+    else:
+        return render_template("storeAnon.html", store_uuid=store_uuid, latitude=latitude, longitude=longitude,
+            display_name=display_name, strikes=strikes, safety_ratings=safety_ratings,
+            user_submissions=user_submissions, store_submissions=store_submissions)
 
 
-    return render_template("store.html", store_uuid=store_uuid, latitude=latitude, longitude=longitude,
-                           display_name=display_name, strikes=strikes, safety_ratings=safety_ratings,
-                           user_submissions=user_submissions, store_submissions=store_submissions, form=form)
+    
 
 
 @app.route('/user/<username>', methods=['GET', 'POST'])
